plugins/CoreConsole/Commands/GeneratePluginBase.php

```diff
@@ -397,4 +404,10 @@ private function replaceContent($replace, $contentToReplace)
         return $contentToReplace;
     }
 
+    protected function throwErrorIfNotGitInstalled()
+    {
+        if (!SettingsPiwik::isGitDeployment()) {
+            throw new NotGitInstalledException('Using this feature Matomo has to be installed through git!');
```

Comment:
Maybe we could improve this message a bit and also mention the url `https://developer.matomo.org/guides/getting-started-part-1``, which should give more details how to set up for development.

Reply:
Good point, I'll do it.

Reply:
@sgiehl I added the url to the message, but not sure about how it could be improved. any recommendation?

Currently it is:
`Using this feature Matomo has to be installed through git. For more information please visit https://developer.matomo.org/guides/getting-started-part-1.`

Reply:
I'm not sure if the `installed through git` is that helpful. Downloading the package from GitHub might actually include the require plugin files, but there won't be a `.git/HEAD`, so the command would still raise an error. So actually it would be required to do a full git checkout. Guess it doesn't hurt if the message is a bit more technical, as it actually should be used by developers only.

So maybe something like `This development feature requires Matomo to be checked out from git. For more information please visit https://developer.matomo.org/guides/getting-started-part-1.`